### PR TITLE
Chore: Split get user by ID

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
Splits the Get User by ID method and replaces and removes the old one from sqlstore. Fix for enterprise: https://github.com/grafana/grafana-enterprise/pull/3632.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mildwonkey — COMMENTED** [review]
the code looks great! 

I don't understand what's going on with the wire changes - that's not your fault; I'm now concerned I did this part wrong on my PRs. I added services to the `wireBasicSet` only. 

I don't feel confident approving this without a second +1 on the wire changes (for my own edification), but overall this looks 👍 from my perspective!

**Comment by idafurjes** [comment]
@mildwonkey the reason why I adjusted the runners wire file is because I am injected user service into the commands and the commands and runner have a separate wire file, thanks for the review @mildwonkey I will fix the things we discussed

**Review by zserge — APPROVED** [review]

File: pkg/services/login/authinfoservice/database/database.go [hunk 1]
```diff
@@ -18,13 +19,15 @@ type AuthInfoStore struct {
 	sqlStore       sqlstore.Store
 	secretsService secrets.Service
 	logger         log.Logger
+	userService    user.Service
 }
 
-func ProvideAuthInfoStore(sqlStore sqlstore.Store, secretsService secrets.Service) *AuthInfoStore {
+func ProvideAuthInfoStore(sqlStore sqlstore.Store, secretsService secrets.Service, userService user.Service) login.Store {
```
[the remark this answers is not in the document]
Comment (idafurjes) [reply pkg/services/login/authinfoservice/database/database.go:25]: in the new services, we always return and interface with provide service

[the remark this answers is not in the document]
Comment (papagian) [reply pkg/services/login/authinfoservice/database/database.go:25]: yes but what is the reason for that?

**Pull request merged**